pkg/drain/nodegroup.go

```diff
@@ -27,6 +29,13 @@ import (
 // retryDelay is how long is slept before retry after an error occurs during drainage
 const retryDelay = 5 * time.Second
 
+var recoverablePodEvictionErrors = [...]string{
```

Comment:
This list may not be exhaustive.  These are the ones I found through manual testing.

Reply:
Hmm, I wonder if we could check more reliably than this, using some Kubernetes `const` for the error message. I'm a bit reluctant to rely on error messages that are subject to change, trimming, extra spaces, etc...

Reply:
```
	"TooManyRequests",
	"NotFound",
```

These two are definitely meta error `	StatusReasonNotFound StatusReason = "NotFound"` and `StatusReasonTooManyRequests StatusReason = "TooManyRequests"`.

The other two appear to be some aws specific things. How did you encounter them? Can you print the whole error so we can see what interface it implements at that point?

Reply:
I've updated this to use the checker functions built into apimachinery.

I've re-ran a manual test and it works perfectly.